src/components/Cards/assignee-badge.tsx

```diff
@@ -1,29 +1,131 @@
+import { SyntheticEvent, useState } from "react";
 import Image from "next/image";
+import { useRouter } from "next/router";
+import { RepairStatus } from "@prisma/client";
+
+import Button from "@/components/Button";
+import Circle from "@/components/Cards/circle";
+import VolunteerCard from "@/components/Cards/volunteer-card";
+import Modal from "@/components/Modal";
+import { useRepairers } from "@/hooks/events";
+import { useUpdateRepairRequest } from "@/hooks/repair-request";
 
 type Props = {
-  firstName?: string;
-  lastName?: string;
-  avatar?: string;
+  firstName: string | undefined;
+  lastName: string | undefined | null;
+  avatar?: string | undefined;
+  assignedTo?: string;
+  repairRequestId?: string;
 };
 
-export default function AssigneeBadge({ firstName, lastName, avatar }: Props) {
+export default function AssigneeBadge({
+  firstName,
+  lastName,
+  avatar,
+  assignedTo,
+  repairRequestId
+}: Props) {
+  const [showAssigneeModal, setShowAssigneeModal] = useState(false);
+  const {
+    query: { id: eventId }
+  } = useRouter();
+  const { data: repairers } = useRepairers(eventId as string);
+  const { mutate: updateRepairRequest } = useUpdateRepairRequest(
+    repairRequestId as string
```

Comment:
Lets remove this casting. It looks like repairRequestId should not be optional at all.